draft-ietf-tls-tls13.md

```diff
@@ -2969,24 +2977,27 @@ MUST generate a fatal "unexpected_message" alert.  After sending a
 KeyUpdate message, the sender SHALL send all its traffic using the
 next generation of keys, computed as described in
 {{updating-traffic-keys}}. Upon receiving a KeyUpdate, the receiver
-MUST update their receiving keys and if they have not already updated
-their sending state up to or past the then current receiving
-generation MUST send their own KeyUpdate prior to sending any other
-messages.  This mechanism allows either side to force an update to the
-entire connection. Note that implementations may receive an arbitrary
+MUST update its receiving keys.
+
+If the requestUpdate field is set to "true" then the receiver MUST
+send a KeyUpdate of its own with requestUpdate set to "false" prior
+to sending its next application data record. This mechanism allows either side to force an update to the
+entire connection, but causes an implementation which
+receives multiple KeyUpdates while it is silent to respond with
+a single update. Note that implementations may receive an arbitrary
 number of messages between sending a KeyUpdate and receiving the
 peer's KeyUpdate because those messages may already be in flight.
-
-Note that if implementations independently send their own
-KeyUpdates and they cross in flight, this only results in an
-update of one generation; when each side receives the other
-side's update it just updates its receive keys and notes that
-the generations match and thus no send update is needed.
-
-Note that the side which sends its KeyUpdate first needs to retain
+The side which sends its KeyUpdate first needs to retain
```

Comment:
Do we even need this section with the ladders split? It seems just needless mess to recommend people drop the traffic secret and not the receive traffic keys like that. I think I would drop the sentence beginning with "The side which sends its [...]" and then maybe stick a "However" in front of "Because send and receive [...]" to transition from "Note that implementations may [...]"